### ISSUE TITLE
How to set effect for IKEA Tradfri LED1624G9

### ISSUE THREAD

**Issue by TheBlubb14** [ioBroker/ioBroker.zigbee#1216]
On the [zigbee2mqtt site](https://www.zigbee2mqtt.io/devices/LED1624G9.html#effect-enum) is listed that my bulb has effect support. How can I trigger such effects from within ioBroker?
I tried to add a custom state `effect` to my bulb in the objects tab. When trying to set the effect that way I get the following error `SendToZigbee failed! (Error: Unknown attribute 'effect', specify either an existing attribute or a number)`

Below current device info. Updated to the latest bulb firmware and have `ioBroker.zigbee` version `1.5.6` installed

![grafik](https://user-images.githubusercontent.com/12298942/133909258-3439e30e-41dd-4416-a865-381bf23d0552.png)

**Comment by asgothian** [comment]
Adding a state in the objects tab will not offer access to the effect. I am not sure the bulb supports effects. Please verify this with zigbee2mqtt.io.

A.

**Comment by TheBlubb14** [comment]
> 
> 
> Adding a state in the objects tab will not offer access to the effect. I am not sure the bulb supports effects. Please verify this with zigbee2mqtt.io.
> 
> A.

It does as seen here https://www.zigbee2mqtt.io/devices/LED1624G9.html#effect-enum

**Comment by asgothian** [comment]
Please add the device to the "exclude" or "ausschliessen" tab and restart the zigbee adapter. If the herdsman implementation you are using is supporting effect, you should get a state to control them

A.

**Comment by TheBlubb14** [comment]
> 
> 
> Please add the device to the "exclude" or "ausschliessen" tab and restart the zigbee adapter. If the herdsman implementation you are using is supporting effect, you should get a state to control them
> 
> A.

That worked, thanks.
So the `ioBroker.zigbee` adapter has its own default mapping to the herdsman implementation?
The adapter only validated the states against his own implementation? And by excluding the device type the adapter does not check against that?
The `effect` object will not getting auto added to the other bulbs, right?

**Comment by TheBlubb14** [comment]
> 
> The `effect` object will not getting auto added to the other bulbs, right?

Just updated my other bulbs and the `effect` was added automatically

**Comment by asgothian** [comment]
There may be a negative side effect - the bulbs will no longer turn on automatically when brightness is moved from 0 to a higher number. Same with off.. the state may remain on even if brightness is lower to 0. The herdsman implementation does not do that yet and the stuff we did in the zigbee adapter does not translate to the new method.

as far as implementation goes. It used to be that the herdsman only provides messages, and the handling of those messages needed to be explicitly implemented in the zigbee adapter. Starting with zigbee 1.4.1, we were able to rely on a new option from the herdsman-converters called exposes, which tells the adapter which states to create. Any device with a dedicated implementation in the zigbee adapter will keep using that and ignore the exposes. If such a device is added to the exclude list., the adapter will ignore the dedicated implementation and use the exposes instead. Thats what happened in your case.

A.
